typing/ctype.ml

```diff
@@ -408,15 +404,15 @@ let in_subst_mode = function
   | Pattern _ -> false
 
 let can_generate_equations = function
-  | Expression _ | Pattern { equations_generation = Forbidden } -> false
-  | Pattern { equations_generation = Allowed _ } -> true
+  | Expression _ -> false
+  | Pattern _  -> true
```

Comment:
The `can_generate_equations` function is now an alias for `in_pattern_mode`. I think it might be clearer to remove the function. 
At the very least, the redundant checks on `if in_pattern_mode uenv then (... if can_generate_equations then ...)` should be removed in `unify3_var` and `unify3`.